apps/heft/src/cli/HeftToolsCommandLineParser.ts

```diff
@@ -82,4 +158,21 @@ export class HeftToolsCommandLineParser extends CommandLineParserBase {
       this._pluginManager.initializePlugin(pluginSpecifier);
     }
   }
+
+  private async _reportErrorAndSetExitCode(error: Error): Promise<void> {
+    this.terminal.writeErrorLine(error.toString());
+
+    if (this.isDebug) {
+      this._terminal.writeLine();
+      this._terminal.writeErrorLine(error.stack!);
+    }
+
+    await this._metricsCollector.flushAndTeardownAsync();
+
+    if (!process.exitCode || process.exitCode > 0) {
+      process.exit(process.exitCode);
```

Comment:
We should eliminate `process.exit()` from our tools. It is a deprecated practice, because it will forcibly kill any unresolved promises.

Reply:
Agreed - currently Rush does this too. Let's tackle this later.